Client/Frontend/Browser/HomePanel/BackgroundImage.swift

```diff
@@ -0,0 +1,131 @@
+            // Do absolutely nothing
+            self.info = nil
+            return
+        }
+        
+        // Setting up normal backgrounds
+        if BackgroundImage.standardBackgrounds == nil {
+            BackgroundImage.standardBackgrounds = BackgroundImage.generateStandardData(file: backgroundFilePath)
+        }
+        
+        // Setting up sponsored backgrounds
+        if BackgroundImage.sponsors == nil && Preferences.NewTabPage.backgroundSponsoredImages.value {
+            BackgroundImage.sponsors = BackgroundImage.generateSponsoredData(file: sponsoredFilePath)
+        }
+        
+        self.info = BackgroundImage.randomBackground()
+    }
+    
+    private static func generateSponsoredData(file: String) -> [Background] {
+        guard let json = BackgroundImage.loadImageJSON(file: file),
+            let region = NSLocale.current.regionCode else {
+                return []
+        }
+        
+        let dateFormatter = DateFormatter().then {
+            $0.locale = Locale(identifier: "en_US_POSIX")
+            $0.dateFormat = "yyyy-MM-dd'T'HH:mm:ssZZZZZ"
+            $0.calendar = Calendar(identifier: .gregorian)
+        }
+        
+        let today = Date()
+        // Filter down to only regional supported items
+        let regionals = json.filter { ($0["regions"] as? [String])?.contains(region) == true }
+        
+        // Filter down to sponsors that fit the date requirements
+        let live = regionals.filter { item in
+            guard let dates = item["dates"] as? [String: String],
+                let start = dateFormatter.date(from: dates["start"] ?? ""),
+                let end = dateFormatter.date(from: dates["end"] ?? "") else {
+                    return false
+            }
+            
+            return today > start && today < end
+        }
+        
+        return generateBackgroundData(data: live)
+    }
+    
+    private static func generateStandardData(file: String) -> [Background] {
+        guard let json = BackgroundImage.loadImageJSON(file: file) else { return [] }
+        return generateBackgroundData(data: json)
+    }
+    
+    private static func generateBackgroundData(data: [[String: Any]]) -> [Background] {
+        return data.compactMap { item in
+            guard let image = item["image"] as? String,
+                let center = item["center"] as? CGFloat else { return nil }
+            
+            if let credit = item["credit"] as? [String: String],
+                let name = credit["name"] {
+                return Background(imageFileName: image, center: center, credit: (name, credit["url"]))
+            }
+            
+            return Background(imageFileName: image, center: center, credit: nil)
+        }
+    }
+    
+    private static func randomBackground() -> Background? {
+        // Determine what type of background to display
+        let sponsorshipShowRate = 4 // e.g. 4 == 25%
```

Comment:
This is an important constant and is buried down there.
I would move it somewhere higher for easy adjustments later